src/use-auto-save.test.js

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from "react";
+import { render, fireEvent, waitForElement } from "react-testing-library";
+import useAutoSave from "./use-auto-save";
+
+// ---------------------------
+// Mocks
+// ---------------------------
+
+const MockSaveComponent = ({ value, save, delay }) => {
+  useAutoSave(value, save, delay);
+
+  return <div>{value}</div>;
+};
+
+const MockCleanup = ({ value, save, delay }) => {
+  const [show, setShow] = useState(true);
+
+  return (
+    <>
+      {show ? (
+        <MockSaveComponent value={value} save={save} delay={delay} />
+      ) : null}
+      <button onClick={() => setShow(!show)}>Toggle AutoSave</button>
+    </>
+  );
+};
```

Comment:
What if we use [`unmount`](https://testing-library.com/docs/react-testing-library/api#unmount) from `react-testing-library` in our tests instead of using this component? May make our tests clearer.

Reply:
So much better!